src/app/common/TagList.vue

```diff
@@ -66,10 +65,15 @@ function getRoute(tag: LabelValue): RouteLocation | undefined {
         })
       }
       case 'kuma.io/service': {
+        // Annotations by themselves don’t have information about a service’s associated mesh. The easiest solution is to read this information from the current route. A better approach could be to provide the current mesh as an optional prop to `TagList`.
+        if (!('mesh' in route.params)) {
+          return undefined
+        }
+
```

Comment:
agree that the better approach would to provide this from The Outside, so thanks for the comment here, should we add a TODO on the front?

Not totally aware of how TagList currently works though, but at a brief look we could get rid of the entire switch statement by providing a possible href for tags from The Outside.

Anyway thanks again for the comment, definitely one to avoid working on here 👏 We should follow up at some point though.

Reply:
The only way to be able to provide the links up front would be to have this exact logic on the outside. That logic should really be here so we can always rely on it. Only then can we just show tags for the annotations coming from our resources without thinking about it.

That being said, it makes sense to let the consuming side provide a mesh value if one is available: it seems logical that only mesh-specific resources have annotation referring to mesh resources. I’ll leave a comment.